main.js

```diff
@@ -1,24 +1,105 @@
 #!/usr/bin/env node
+
+/*
+	Run by default in interactive mode. When called in script mode, process.exit() should be called in your script to exit the nodejs app.
+	Arguments:
+	- a path which target a JavaScript file to execute (.js extension).
+	- a path which target an ipc path.
+*/
+
 var Web3 = require('web3');
 var web3admin = require('./web3Admin.js');
 var repl = require('repl');
 var net = require('net');
+var fs = require('fs')
+var vm = require('vm');
 var ipcpath = require('./getIpcPath.js');
 
+var ipcPath = ipcpath();
+var jsScript;
+var help = false;
+if (!processArguments())
+	return;
+
+if (help)
+{
+	logHelp();
+	return;
+}
+
 process.on('uncaughtException', function(err) {
-    console.error("Uncaught exception: " + err);
+	console.error("Uncaught exception: " + err);
 });
 
-console.log("Connecting to node at " + ipcpath());
-web3 = new Web3(new Web3.providers.IpcProvider(ipcpath(), net));
+console.log("Connecting to node at " + ipcPath);
+web3 = new Web3(new Web3.providers.IpcProvider(ipcPath, net));
 web3admin.extend(web3);
-web3.eth.getBlockNumber(function(err, number) {
-    if (err) {
-        console.error("Could not connect to node. Please start an Ethereum node first.");
-    } else {
-        console.log("Connection successful.");
-        console.log("Current block number: " + number); 
-        console.log("Entering interactive mode.");
-        var replServer = repl.start({});
-    }
+web3.eth.getBlockNumber(function(err, number)
+{
+	if (err)
+		console.error("Could not connect to node. Please start an Ethereum node first.");
+	else
+	{
+		console.log("Connection successful.");
+		console.log("Current block number: " + number);
+		if (jsScript)
+			executeScript();
+		else
+		{
+			console.log("Entering interactive mode.");
+			repl.start({});
+		}
+	}
 });
+
+function processArguments()
+{
+	var notRecognized = false;
+	for (var k = 2; k < process.argv.length; k++)
+	{
+		var arg = process.argv[k];
+		if (arg.endsWith('.js'))
+			jsScript = arg;
+		else if (arg === "help" || arg === "--help" || arg === "-h")
+			help = true;
+		else if (arg.startsWith("ipc:") || arg.endsWith(".ipc"))
+			ipcPath = arg.startsWith("ipc:") ? arg.substring(4) : arg;
+		else
+		{
+			notRecognized = true;
+			console.log("Argument not recognized " + arg);
+		}
+	}
+	if (notRecognized)
+	{
+		logHelp();
+		return false;
+	}
+	return true;
+}
+
+function executeScript()
+{
+	console.log("Executing " + jsScript + " ...");
+	fs.readFile(jsScript, 'utf8', function (err, data)
+	{
+		if (err)
+		{
+			console.log(err);
+			process.exit();
+		}
+		else
+		{
+			var script = new vm.Script(data);
+			script.runInThisContext();
+		}
+	});
+}
+
+function logHelp()
+{
+	console.log("Run by default in interactive mode. When called in script mode, process.exit() should be called in your script to exit the nodejs app.")
```

Comment:
We should probably also tell the people what this is. Please also avoid long lines. Perhaps like that:
```
Usage: ethconsole [javascript file] [ipc socket]
Connects to an ethereum node via ipc in order to control it remotely
via access to the global variable `web3` (the admin interface is also present).
If no arguments are given, connects to the standard ipc socket at <INSERT STANDARD SOCKET HERE>
and drops into interactive mode.
Arguments:
  <ipc socket path>  connect to the given ipc socket (use ipc://<path> if it does not end with .ipc)
  <javascript file>    execute the given javascript file that has to end in .js non-interactively.
                           The script has to call process.exit() in order to terminate the console.
```